### PR TITLE
Use java-action for tests

### PR DESCRIPTION
**Pull request by therve**
Don't use jabba which is slow to install.

### CODE REVIEW

File: .github/workflows/approved_status.yml [hunk 1]
```diff
@@ -26,10 +26,8 @@ jobs:
           app_id: ${{ secrets.PIPELINE_GITHUB_APP_ID }}
           private_key: ${{ secrets.PIPELINE_GITHUB_APP_PRIVATE_KEY }}
           repository: DataDog/datadog-api-spec
-      - name: Checkout code
-        uses: actions/checkout@v2
       - name: Post PR review status check
         uses: DataDog/github-actions/post-review-status@v1.0.0
-        with:
+        wit,:
```
Comment (therve) [inline .github/workflows/approved_status.yml:31]:
```suggestion
        with:
```

File: .github/workflows/test.yml [hunk 2]
```diff
@@ -81,35 +80,33 @@ jobs:
...
+      - uses: actions/checkout@v3
       - name: Install Java
-        # the setup-scala action is the only available action that allows us to access
-        # https://github.com/shyiko/jabba - a tool that can install various JDKs
-        uses: olafurpg/setup-scala@v10
+        uses: actions/setup-java@v3
         with:
-          java-version: openjdk@1.16.0-1
-          jabba-version: 0.11.2
+          java-version: '16.0.1'
```
Comment (therve) [inline .github/workflows/test.yml:107]:
```suggestion
          java-version: '16'
```

File: .github/workflows/test.yml [hunk 3]
```diff
@@ -81,35 +80,33 @@ jobs:
       matrix:
         # see DEVELOPMENT.md for information why we're using this specific combination
         # of JDKs to run unit tests on
-        java-version: ["adopt@1.8.0-292", "openjdk@1.16.0-1"]
+        java-version: ["8", "16.0.1"]
```
Comment (therve) [inline .github/workflows/test.yml:83]:
```suggestion
        java-version: ["8", "16"]
```

File: .github/workflows/test_integration.yml [hunk 4]
```diff
@@ -51,12 +51,11 @@ jobs:
...
           context: integration
       - name: Install Java
-        # the setup-scala action is the only available action that allows us to access
-        # https://github.com/shyiko/jabba - a tool that can install various JDKs
-        uses: olafurpg/setup-scala@v10
+        uses: actions/setup-java@v3
         with:
-          java-version: openjdk@1.16.0-1
-          jabba-version: 0.11.2
+          java-version: '16.0.1'
```
Comment (therve) [inline .github/workflows/test_integration.yml:56]:
```suggestion
          java-version: '16'
```

File: .github/workflows/prepare_release.yml [hunk 5]
```diff
@@ -34,7 +34,8 @@ jobs:
         uses: actions/setup-java@v2
```
Comment (skarimo) [inline .github/workflows/prepare_release.yml:34]:
```suggestion
        uses: actions/setup-java@v3
```

**Review by skarimo — COMMENTED** [review]
Small nit to use same version everywhere

**Review by skarimo — APPROVED** [review]

**Pull request merged**